File: app/src/main/java/com/cocobiz/app/domain/model/UserProfile.kt
package com.cocobiz.app.domain.model

data class UserProfile(
    val id: Long = 1L,
    val businessName: String = "",
    val ownerName: String = "",
    val phone: String = "",
    val alternatePhone: String = "",
    val email: String = "",
    val address: String = "",
    val city: String = "",
    val state: String = "",
    val pincode: String = "",
    val gstNumber: String = "",
    val logoPath: String = "",
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis()
) {
    val isComplete: Boolean
        get() = businessName.isNotBlank() && ownerName.isNotBlank() && phone.isNotBlank()
}

File: backend/models/UserProfile.js
const mongoose = require('mongoose');

const userProfileSchema = new mongoose.Schema({
  localId: { type: Number, default: 1, unique: true },
  businessName: { type: String, default: '' },
  ownerName: { type: String, default: '' },
  phone: { type: String, default: '' },
  alternatePhone: { type: String, default: '' },
  email: { type: String, default: '' },
  address: { type: String, default: '' },
  city: { type: String, default: '' },
  state: { type: String, default: '' },
  pincode: { type: String, default: '' },
  gstNumber: { type: String, default: '' },
  logoPath: { type: String, default: '' },
  createdAt: { type: Number, default: Date.now },
  updatedAt: { type: Number, default: Date.now }
});

module.exports = mongoose.model('UserProfile', userProfileSchema);
